src/node_buffer.cc

```diff
@@ -1142,6 +1142,143 @@ void IndexOfNumber(const FunctionCallbackInfo<Value>& args) {
                                 : -1);
 }
 
+void NoopCallback(char* data, void* hint) {
+  // Intentional no-op
+}
+
+void Split(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  THROW_AND_RETURN_UNLESS_BUFFER(env, args[0]);
+  SPREAD_ARG(args[0], js_buf);
+  SPREAD_ARG(args[1], js_sep_buf);
+
+  std::vector<size_t> index_arr;
+  Local<Object> obj;
+  Persistent<Object> persistent(env->isolate(), obj);
+  Local<v8::Array> res_buffers;
+  size_t cursor = 0;
+  size_t len = 0;
+
+  // Rationale: Find all occurences of the buffer bytes, store those index and
+  // copy bytes from index bounds to new array of buffers
+
+  // REVIEW(eljefedelrodeodeljefe): copy first necessary?
+  char* buf = new char[js_buf_length];
+  memcpy(buf, js_buf_data, js_buf_length);
+
+  size_t sep_buf_len = js_sep_buf_length;
+  char* sep_buf = new char[sep_buf_len];
+  memcpy(sep_buf, js_sep_buf_data, sep_buf_len);
+
+  unsigned char first_match_byte = *((unsigned char *)&sep_buf + 0);
+  /* emtpy buffer separator? Then skip indexing and fancy mapping! */
+  if (sep_buf_len == 0)
+    goto map_bytes;
+
+  for (size_t i = 0; i < js_buf_length; ++i) {
+    unsigned char byte = *((unsigned char *)&buf + i);
+    // check first occurence...
+    // ...and if separator > 1 check consecutive bytes for equalness.
+    if (sep_buf_len == 1 && byte == first_match_byte) {
+      index_arr.push_back(i);
+    } else if (byte == first_match_byte) {
+      for (size_t j = 0; j < sep_buf_len; ++j) {
+        unsigned char sep_byte = *((unsigned char *)&buf + i + j);
+        // only push to array when the whole separator got compared
+        if (sep_byte == sep_buf[j] && j == sep_buf_len - 1) {
+          index_arr.push_back(i);
+        }
+      }
+    }
+  }
+
+  res_buffers = v8::Array::New(env->isolate(), index_arr.size());
```

Comment:
If you want to take advantage of this it's `env->push_values_to_array_function()`. Can see the usage of it in `node::GetActiveRequests()` in `src/node.cc`.